docs/building-extension-modules.md

```diff
@@ -0,0 +1,338 @@
+---
+title: "Building extension modules"
+draft: false
+type: docs
+layout: single
+
+menu:
+  docs:
+    weight: 125
+---
+
+# Building Extension Modules
+
+{{% warning %}}
+While this feature has been around since almost the beginning of the Poetry project and has needed minimal changes,
+it is still considered unstable. You can participate in the discussions about stabilizing this feature
+[here](https://github.com/python-poetry/poetry/issues/2740).
+
+And as always, your contributions towards the goal of improving this feature are also welcome.
+{{% /warning %}}
+
+Poetry allows a project developer to introduce support for, build and distribute native extensions within their project.
+In order to achieve this, at the highest level, the following steps are required.
+
+{{< steps >}}
+{{< step >}}
+**Add Build Dependencies**
+
+The build dependencies, in this context, refer to those Python packages that required in order to successfully execute
+your build script. Common examples include `cython`, `meson`, `maturin`, `setuptools` etc., depending on how your
+extension is built.
+
+{{% note %}}
+You must assume that only python built-ins are available by default in a build environment. This means, if you use need
+even packages like `setuptools`, it must be explicitly declared.
+{{% /note %}}
+
+The necessary build dependencies must be added to the `build-system.requires` section of your `pyproject.toml` file.
+
+```toml
+[build-system]
+requires = ["poetry-core", "setuptools", "cython"]
```

Comment:
Do we want to use unrestricted build requirements here or do we want to set upper/lower bounds (cf #9812)?

Reply:
Well that is up to the user. And honestly keeping this doc up to date every time we release core is a burden without much gain. But I am not firm on best approach here. Not sure which is more likely: someone copy pasting outdated version of this or someone being bitten by a major version upgrade breakage.

Reply:
I have added a note about this in the doc.